### ISSUE TITLE
Are there plans for a python version?

### ISSUE THREAD

**Issue by RijulGupta-DM** [NerdsWithAttitudes/WarpFactory#1]
Curious if a python-based version is on the roadmap. If not, I might take a crack at it!

**Comment by joaopedrosdmm** [comment]
Come here to comment the same. You are losing potential contributors by offering only a matlab version. I still have the Matlab from college, but for those who don't have access to it, it is very expensive.

**Comment by pbbp0904** [comment]
I agree. MATLAB's GPU capabilities we found were very good. Very easy to implement and very fast (much faster than using C on the CPU) but its pricing is unfortunately a big downside for many people who aren't associated with a university/equivalent. We figured we'd post the code and if there's enough interest, a python version could be created. Leaving this thread open for more comments.

**Comment by tkelley3** [comment]
Visualizations in python always seem like a pain point. I can see the advantages of it being written in MATLAB. As an academic, I have an advantage of a cost-free site license for MATLAB. 
 But I would also be interested in the challenge of translating to python.

**Comment by vitasam** [comment]
+1 for the Python (CUDA) version

**Comment by bugsiesegal** [comment]
Going to try porting it myself.

**Comment by DragonlordNewb** [comment]
As someone who uses software exactly like this very frequently, a python version would be much appreciated.

**Comment by evsergey** [comment]
Has anybody started to port the project?

**Comment by yakman2020** [comment]
> I came here to comment the same. You are losing potential contributors by offering only a matlab version. I still have the Matlab from college, but for those who don't have access to it, it is very expensive.

Actually, one can use Octave which is usually pretty compatible with matlab. Not to say that a numpy version wouldn't be a good idea. In fact I'd be happy to help if there was a dev branch, fork or whatever.

**Comment by JohnsterID** [comment]
python attempt: https://github.com/JohnsterID/WarpFactory

**Comment by pbbp0904** [comment]
> python attempt: https://github.com/JohnsterID/WarpFactory

Looks like this is coming along well! The Tricky part will be getting a general solver, not just for specific metrics. But LLMs are pretty good these days so it might not be too bad using their assistance.

**Comment by pseudotensor** [comment]
Another python attempt: https://github.com/pseudotensor/WarpFactoryPython
Using claude code with sonnet-4-5.
